src/main/java/com/amazon/ion/MacroAwareIonReader.kt

```diff
@@ -11,9 +11,27 @@ import java.io.IOException
 interface MacroAwareIonReader : Closeable {
 
     /**
-     * Performs a macro-aware transcode of the stream being read by this reader.
+     * Performs a macro-aware transcode of all values in the stream. This is
+     * shorthand for calling [prepareTranscodeTo], then calling [transcodeNext]
+     * repetitively until it returns `false`.
+     * @param writer the writer to which the reader's stream will be transcoded.
+     */
+    @Throws(IOException::class)
+    fun transcodeAllTo(writer: MacroAwareIonWriter)
+
+    /**
+     * Prepares the reader to perform a macro-aware transcode to the given
+     * writer. This must be called before calling [transcodeNext], but is not
+     * necessary if calling [transcodeAllTo].
+     * @param writer the writer to which the reader's stream will be transcoded.
+     */
+    fun prepareTranscodeTo(writer: MacroAwareIonWriter)
```

Comment:
What makes the `prepareTranscodeTo(writer)`/`transcodeNext()` preferable to `transcodeTo(writer)`? From a glance over it looks like either would work, so I likely missed something.

Reply:
It's not a huge difference, but the current factoring allows for the setup (registering the IVM callback) to happen once, rather than before every value that is transcoded.

Reply:
That makes sense, I see it now. It's really more of a question of whether you know that the callback has been registered, right? So could that state also be carried implicitly, by virtue of a private method that assumes the callback has already been registered and a public method which does the registration then calls the private method? Or am I misunderstanding how this works?

Reply:
Something like that may be possible. We can follow up in #1015. My main goal with this factoring was to avoid repetitive re-setting of the callback. It's not as easy as "is a callback already registered", because a callback is always registered so that the higher-level reader gets notified when IVMs are encountered. Here, we augment that functionality.